gui/src/main.rs

```diff
@@ -137,10 +139,19 @@ impl App for Gui {
                 ui.label("URL:");
                 ui.text_edit_singleline(&mut self.config.url);
                 if ui.button("Submit").clicked() {
+                    let body = if self.config.selected_http_method != HttpMethod::GET {
+                        Some(serde_json::from_str(&self.config.body_str).expect("Body is invalid json"))
```

Comment:
Need to still gracefully handle this invalid parsing, if it fails it panics. the .expect() just leaves a nicer message in the terminal.